app/events/domains/proxy_domains_changed_event.rb

```diff
@@ -0,0 +1,29 @@
+require 'uri'
+
+class Domains::ProxyDomainsChangedEvent < BaseEventStoreEvent
+  def self.create(proxy)
+    new(
+      proxy:    proxy,
+      staging_domains: extract_domain(proxy.sandbox_endpoint),
+      production_domains: extract_domain(proxy.endpoint),
+
+      metadata: {
+        provider_id: (provider_id = proxy.provider&.id),
+        zync: {
+          tenant_id: proxy.tenant_id || proxy.provider&.tenant_id || provider_id,
+          service_id: proxy.service_id,
+        }
+      }
+    )
+  end
+
+  def self.valid?(proxy)
+    !!proxy # TODO: check if deployment_option or domains actually changed
+  end
+
+  def self.extract_domain(url)
```

Comment:
I don't think that this should be responsability of the event 🤔 And this is doing more or less that same as:
https://github.com/3scale/porta/blob/49032f8f4e7c80bbb6a37bf0858190ffa0b45eae/app/models/proxy.rb#L301-L309
Only that you want the 2 endpoints separately and that u don't want 'localhost' if they fail... but then i think this needs a refactoring 😄 not necessarily for today though 👍

Reply:
All events build their own structure. Sure, the proxy object could expose `endpoint_domain` and `sandbox_endpoint_domain`.